proposed/activity-improvements.md

```diff
@@ -0,0 +1,290 @@
+
+We expect to achieve this by additions to the Activity object model and a new ActivityListener type. These additions should provide stronger contracts between in-proc producers and consumers of distributed tracing data, a richer set of information that Activity can represent, and streamlined calling patterns for typical scenarios.
+
+## Scenarios and User Experience
+
+#### 1. A developer wants a portion of code to be annotated with a named block in a distributed trace Gantt chart
+
+The developer needs to wrap their region of code in a using statement that creates the Activity and assigns it a name. It is possible that the activity reference will be null if there is nothing in the process listening for this telemetry or if only a sub-set of telemetry is being sampled.
+
+```C#
+    using (Activity? activity = Activity.StartNew("MyLibrary.Subcomponent.FooOperation")
+    {
+        // optionally the activity can be enriched with extra data
+        activity?.AddTag("userId", userId);
+        activity?.AddTag("fooParam", foo);
+            
+        // the developer's original code goes here
+    }
+```
+
+### 2. Monitoring agent wants to listen to activities
+
+A monitoring agent can create a listener which represents their subscription to activity callbacks like this.
+```C#
+
+IDisposable listener = Activity.StartListening(
+           shouldCreateActivity: (name, parentContext, kind, tags, links) => true,
+           onActivityStarted: (activity) => { ... },
+           onActivityStopped: (activity) => { ... });
+
+```
+It is up to the listener to do extract information from the Activity object and respond to the callbacks however they like, probably by serializing a message to a remote service that will aggregate the telemetry.
+
+[TODO]: We could flesh out what we expect it look like end-to-end to use this with OpenTelemetry and perhaps some UI? It would help newcomers understand but I don't think it is a blocking issue for the current set of collaborators.
+
+Requirements:
+=============
+
+### Goals
```

Comment:
is it a goal to keep using `Activity` class? Just want to make sure we are calling this out explicitly.

Reply:
Yes. I consider it a fall out from the desire not to break back-compat and the desire for OpenTelemetry to receive all existing telemetry. Switching to any other type means all the existing instrumentation is lost.

Happy to get that called out.